idaes/core/base/tests/test_extended_control_volume_1d.py

```diff
@@ -0,0 +1,203 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2024 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+Tests for ExtendedControlVolumeBlockData.
+
+Author: Andrew Lee
+"""
+import pytest
+from pyomo.environ import ConcreteModel, Constraint, units
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import (
+    ExtendedControlVolume1DBlock,
+    FlowsheetBlockData,
+    declare_process_block_class,
+)
+from idaes.core.util.exceptions import (
+    ConfigurationError,
+)
+from idaes.core.util.testing import (
+    PhysicalParameterTestBlock,
+)
+
+
+# -----------------------------------------------------------------------------
+# Mockup classes for testing
+@declare_process_block_class("Flowsheet")
+class _Flowsheet(FlowsheetBlockData):
+    def build(self):
+        super(_Flowsheet, self).build()
+
+
+@pytest.mark.unit
+def test_add_isothermal_constraint():
+    m = ConcreteModel()
+    m.fs = Flowsheet(dynamic=False)
+    m.fs.pp = PhysicalParameterTestBlock()
+
+    m.fs.cv = ExtendedControlVolume1DBlock(
+        property_package=m.fs.pp,
+        transformation_method="dae.finite_difference",
+        transformation_scheme="BACKWARD",
+        finite_elements=10,
+    )
+
+    m.fs.cv.add_geometry()
+    m.fs.cv.add_state_blocks(has_phase_equilibrium=False)
+
+    m.fs.cv.add_isothermal_constraint()
+
+    assert isinstance(m.fs.cv.enthalpy_balances, Constraint)
+    assert len(m.fs.cv.enthalpy_balances) == 1  # x==0 is skipped
```

Comment:
I thought the length of enthalpy_balances would be fine_elements - 1. Maybe I am missing something (or the test is failing).

Reply:
The model has not been discretised, so there are only 2 points.

Reply:
Oh! I see now. Didn't catch that--didn't apply transformation.